### PR TITLE
Cancel when running a delete

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
We currently don't cancel a running job when we are deleting a connection.

This PR is cancelling a running job if it is getting deleted.

It is calling the existing cancel activity if the job is running before deleting the connection.

### CODE REVIEW

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java [hunk 1]
```diff
@@ -700,6 +703,12 @@ private void deleteConnectionBeforeTerminatingTheWorkflow() {
    * Set a job as cancel and continue to the next job if and continue as a reset if needed
    */
   private void reportCancelledAndContinueWith(final boolean isReset, final ConnectionUpdaterInput connectionUpdaterInput) {
+    reportCancelled(isReset);
+    resetNewConnectionInput(connectionUpdaterInput);
+    prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
```
Comment (jdpgrailsdev) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java:708]: Do any of these methods throw exceptions, particularly `RuntimeException`, as I know that we like to use that as a way to signal failures?  If so, should this be in a try/finally block to allow one or two of these methods to still fire if the first one or two raises an exception?

Comment (benmoriceau) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java:708]: Most of the method are getter/setter so very unlikely to throw a RuntimeException. The method that can throw are run using `runMandatoryActivity` which catch any exception and set the job in quarantine.

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java [hunk 2]
```diff
@@ -122,6 +122,9 @@ public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws Retr
       }
 
       if (workflowState.isDeleted()) {
+        if (workflowState.isRunning()) {
+          reportCancelled(false);
```
Comment (lmossman) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java:126]: nit: add a log message here indicating that we are cancelling the job due to deleting the connection

**Review by lmossman — APPROVED** [review]
LGTM

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Pull request merged**